src/maus/transformation.py

```diff
@@ -0,0 +1,136 @@
+Think of the EdiSeed as an intermediate data structure.
+
+MAUS is agnostic regarding your EdiSeed. It doesn't care about how your EdiSeed looks like exactly. It just needs to be
+accessible somehow.
+
+In your application you might use a different data model which is de-coupled from EDIFACT and thus also de-coupled from
+the EdiSeed. This is generally a good idea, because it spares your application from the never-ending pain introduced by
+BDEW and edi@energy every 6 months. However, this comes with the price of having to write a more complex transformation.
+
+This module contains methods that are relevant when transforming data between your edi-similar structure ("EdiSeed")
+and the application domain ("non-edifact").
+"""
+import asyncio
+from typing import Any, Awaitable, Dict, List, Mapping, Optional, Protocol, TypeVar
+
+from maus import DataElementValuePool, ValuePoolEntry
+
+EdifactData = TypeVar("EdifactData")
+"""
+the type which you use to model the edifact side of things (e.g. the type of the EdiSeed)
+"""
+ApplicationData = TypeVar("ApplicationData")
+"""
+the type which you use to model the data in your application (e.g. the type of your BO4E, BOneyComb)
+"""
+
+
+class ApplicationEdifactConverter(Protocol[EdifactData, ApplicationData]):
+    """
+    this class describes a converter from the Application Domain to the EDIFACT Domain
+    """
+
+    async def transform_application_to_edi_domain(self, application_data: ApplicationData) -> EdifactData:
+        """
+        Transforms the given application data (e.g. bo4e/boneycomb) into the edifact domain (e.g. ediseed)
+        :param application_data:
+        :return: the converted data
+        """
+
+    async def transform_edi_to_application_domain(self, edifact_data: EdifactData) -> ApplicationData:
+        """
+        Transforms the given edifact data (e.g. ediseed) into the application domain (e.g. bo4e/boneycomb)
+        :param edifact_data:
+        :return: the converted data
+        """
+
+
+DataModel_contra = TypeVar("DataModel_contra", contravariant=True)
+
+
+class Accessor(Protocol[DataModel_contra]):
+    """description of a class that allows to get and set values from a generic data model"""
+
+    def get_value(self, data: DataModel_contra, discriminator: str) -> Optional[Any]:
+        """
+        Gets the value described by the discriminator from the given data
+        :param data: the data to be read
+        :param discriminator: describes _which_ value should be read and returned
+        :return: the value or None
+        """
+
+    def set_value(self, data: DataModel_contra, discriminator: str, value: Any) -> None:
+        """
+        Sets the value that is described by the disciminator in the given data instance.
+        :param value: the value to be set
+        :param data: the data to be modified
+        :param discriminator: describes _which_ value should be set
+        :return: nothing; it modifies the instance of data
+        """
```

Comment:
neu (seit dem letzten review) ist das hier: statt 5 methoden zu übergeben, übergibt man jetzt drei instanzen, die gewissen methoden implementieren müssen. nicht per harter vererbung (abstract class) sondern über den type hint.

gut möglich, dass ich in ein anti-pattern gelaufen bin und man das nicht tun soll.

Die grundsätzclihe Idee war: 
- weniger Argumente (auch wenn ich die Linter-Warnung immer noch ignoriere)
- eine klarere Trennung zwischen Konvertierung und Zugriffen auf eins der beiden Datenmodelle